core/src/main/scala/ai/h2o/sparkling/backend/SparklingGateway.scala

```diff
@@ -0,0 +1,117 @@
+package ai.h2o.sparkling.backend
+
+import java.io.FileInputStream
+import java.net.InetAddress
+import java.security.cert.Certificate
+import java.security.{KeyStore, PrivateKey}
+
+import ai.h2o.sparkling.H2OContext
+import javax.net.ServerSocketFactory
+import javax.net.ssl.{KeyManagerFactory, SSLContext}
+import org.apache.spark.expose.Logging
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.{SparkConf, SparkFiles}
+import py4j.GatewayServer
+
+import scala.io.Source
+import scala.util.Random
+
+object SparklingGateway extends Logging {
+
+  def main(args: Array[String]) {
+    val spark = SparkSession.builder.getOrCreate()
+    val conf = spark.sparkContext.getConf
+    val secret = readSecret(conf)
+    val hc = H2OContext.getOrCreate()
+    val address = InetAddress.getByName("0.0.0.0")
+    val builder = new GatewayServer.GatewayServerBuilder()
+      .javaPort(gatewayPort(conf))
```

Comment:
may it happen that the port is already occupied?

Reply:
Yup, that can happen. The steam admin needs to pick a free port.

We could make this more robust by letting the code automatically select the port, but in that case we need to implement some sort of callback, to let Steam know about the selected port. I'm fine with that, but I would say this is mostly about what Steam prefers, CC @bilcus 

We already discussed this with @bilcus

Reply:
I almost forgot about the port.. I would prefer for SW to automatically select the port (pass 0 or -1 and it should pick a free port). However, callback might not be possible due to networking. Since you have the H2O proxy, could you add h2o-3 REST endpoint with this information?

Reply:
Good idea, I can expose an endpoint Flow Proxy on the driver side which steam can query, does it make sense @bilcus ?

Reply:
Yes, please!

Reply:
Should be done